File: libs/mediaScan/src/VdrRecording.cc
/**
 * ======================== legal notice ======================
 * 
 * File:      VdrRecording.cc
 * Created:   3. Juli 2012, 08
 * Author:    <a href="mailto:geronimo013@gmx.de">Geronimo</a>
 * Project:   libfsScan: mediatypes and filesystem scanning
 * 
 * CMP - compound media player
 * 
 * is a client/server mediaplayer intended to play any media from any workstation
 * without the need to export or mount shares. cmps is an easy to use backend
 * with a (ready to use) HTML-interface. Additionally the backend supports
 * authentication via HTTP-digest authorization.
 * cmpc is a client with vdr-like osd-menues.
 * 
 * Copyright (c) 2012 Reinhard Mantey, some rights reserved!
 * published under Creative Commons by-sa
 * For details see http://creativecommons.org/licenses/by-sa/3.0/
 * 
 * The cmp project's homepage is at http://projects.vdr-developer.org/projects/cmp
 * 
 * --------------------------------------------------------------
 */
#include <VdrRecording.h>
#include <StringBuilder.h>
#include <File.h>
#include <string.h>
#include <stdlib.h>
#include <stdio.h>
#define FILE_MASK       "/%05d.ts"

cVdrRecording::cVdrRecording(const cFile &File)
 : cAbstractMultiFileMovie(File, "video/mpeg", VdrRecording)
{
  cStringBuilder sb;
  cFile *parent = File.Parent();

  sb.Append(parent->Name());
  sb.Append(" (").Append(File.Name()).Append(")");
  delete parent;
  SetName(sb.toString());
  bufSize = 10;
}

cVdrRecording::~cVdrRecording()
{
}

void cVdrRecording::Refresh(void)
{
  size_t total = 0;
  cFile *tmp;

  if (!checkBuffer()) return;
  movieFiles = 0;
  for (int fileNo = 1;; ++fileNo) {
      sprintf(buf, FILE_MASK, fileNo);
      tmp = new cFile(KeyPath(), buf);

      if (!tmp || !tmp->Exists()) {
         movieFiles = fileNo - 1;
         delete tmp;
         break;
         }
      total += tmp->Size();
      delete tmp;
      }
  SetSize(total);
}

const char *cVdrRecording::FirstFile(void)
{
  if (!checkBuffer()) return NULL;
  curFileNo = 1;
  sprintf(buf, FILE_MASK, curFileNo);
  cFile *tmp = new cFile(KeyPath(), buf);
  const char *rv = NULL;

  if (tmp) {
     rv = tmp->AbsolutePath();
     delete tmp;
     }
  return rv;
}

const char *cVdrRecording::NextFile(void)
{
  if (++curFileNo < movieFiles) {
     if (!checkBuffer()) return NULL;
     sprintf(buf, FILE_MASK, curFileNo);
     cFile *tmp = new cFile(KeyPath(), buf);
     const char *rv = NULL;

     if (tmp) {
        rv = tmp->AbsolutePath();
        delete tmp;

        return rv;
        }
     }
  return NULL;
}

File: cmpcj/src/de/schwarzrot/media/domain/VdrRecording.java
/**
 * ======================== legal notice ======================
 * 
 * File:      VdrRecording.java
 * Created:   13. June 2012, 04:57
 * Author:    <a href="mailto:geronimo013@gmx.de">Geronimo</a>
 * Project:   cmpc - a java frontend (client) part of compound media player
 *                   uses external players to play the media
 * 
 * CMP - compound media player
 * 
 * is a client/server mediaplayer intended to play any media from any workstation
 * without the need to export or mount shares. cmps is an easy to use backend
 * with a (ready to use) HTML-interface. Additionally the backend supports
 * authentication via HTTP-digest authorization.
 * cmpc is a client with vdr-like osd-menues.
 * 
 * Copyright (c) 2012 Reinhard Mantey, some rights reserved!
 * published under Creative Commons by-sa
 * For details see http://creativecommons.org/licenses/by-sa/3.0/
 * 
 * The cmp project's homepage is at http://projects.vdr-developer.org/projects/cmp
 * 
 * --------------------------------------------------------------
 */
package de.schwarzrot.media.domain;


import java.io.File;


public class VdrRecording extends Movie {
    private static final long serialVersionUID = 713L;
    protected static final String DEFAULT_MIME_TYPE = "video/mpeg";
    protected static final String KEY_FILE = "00001.vdr";


    public VdrRecording(Genre parent, File path) {
        this(parent, SupportedMediaType.VdrRecording, DEFAULT_MIME_TYPE, path);
    }


    protected VdrRecording(Genre parent, SupportedMediaType type, String mimeType, File path) {
        super(parent, type, mimeType, path);
    }


    @Override
    public String getName() {
        return getRealPath().getParentFile().getName();
    }


    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder(getName());

        sb.append(" (").append(super.getName()).append(")");

        return sb.toString();
    }


    public static boolean isKeyFile(File file2Check) {
        return file2Check.getName().equals(KEY_FILE);
    }
}
